Configurations/unix-Makefile.tmpl

```diff
@@ -1269,19 +1264,33 @@ tags TAGS: FORCE
 	-ctags -R .
 	-etags `find . -name '*.[ch]' -o -name '*.pm'`
 
-{- output_off() if $disabled{fips}; "" -}
 fips-checksums: generate_fips_sources
-	if which unifdef > /dev/null; then \
-	    ( cd $(SRCDIR) \
-	      && cat providers/fips.module.sources \
-	             | xargs ./util/fips-checksums.sh \
-	             > providers/fips-sources.checksums \
-	      && sha256sum providers/fips-sources.checksums \
-	             > providers/fips.checksum ); \
-	else \
-	    echo >&2 "WARNING: unifdef not in your \$$PATH, FIPS checksums not calculated"; \
+	@which unifdef > /dev/null || \
+	( echo >&2 "ERROR: unifdef not in your \$$PATH, FIPS checksums not calculated"; \
+	  false )
+	( sources=`pwd`/providers/fips.module.sources; \
+	  cd $(SRCDIR) \
+	  && cat $$sources \
+	         | xargs ./util/fips-checksums.sh ) \
+	         > providers/fips-sources.checksums \
+	&& sha256sum providers/fips-sources.checksums \
+	     > providers/fips.checksum
+
+$(SRCDIR)/providers/fips.checksum: providers/fips.checksum
+	cp -p providers/fips.module.sources \
+	    providers/fips-sources.checksums \
+	    providers/fips.checksum $(SRCDIR)/providers
+
+update-fips-checksums: $(SRCDIR)/providers/fips.checksum
```

Comment:
The way this is constructed doesn't account for anyone running this in the source directory, i.e. we're suddenly *forced* to do an out-of-source build when dealing with this stuff.

An option would be to allow different output names to be used, with a make variable...  possibly just one variable that can take a value to be used as a file name prefix.  Then, external scripts (like the github stuff) could simply use that prefix to get different file names to be able to compare between.  That would make the Makefile stuff *much* cleaner.

Reply:
Actually the only thing that does not really work with in-source builds is the diff-fips-checksums. I'd like to merge this as is and there is always room for improvement in subsequent PRs.

Reply:
Thinking about this more - if we just named the files in the builddir with .new, everything would just work even for in-tree builds. I do not think we need a selectable prefix.

Reply:
Why do you need to make "diff-fips-checksums" a make target?

Reply:
Why not? It is convenient.

Reply:
It also Fixes #15139 practically for free.

Reply:
The added `.new` on the intermediate files fixed my immediate concern.